### ISSUE TITLE
DataTable 10.8.4: Drag Selection is still broken

### ISSUE THREAD

**Issue by SolidAnonDev** [primefaces/primereact#7321]
### Describe the bug

Drag Selection on DataTable has been broken for about 2 versions now. Initially my issue #7192 regarding drag selection selecting incorrect records when column filters are enable.

Issue supposedly addressed in #7200 

Issue is no longer causing an error but a selection cannot be made, seems to be a combination of the weird inability to highlight/select text that came up in #7221 

It's also broken on the primereact official site as well, found here:

https://primereact.org/datatable/#multiple_rows_selection

### Reproducer

https://stackblitz.com/edit/vitejs-vite-5pcr8o?file=src%2FApp.tsx

### System Information

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (4) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    primereact: ^10.8.4 => 10.8.4 
    react: ^18.2.0 => 18.2.0
```

### Steps to reproduce the behavior

1. Go to reproducer link and attempt to make a drag selection
2. You will see the behavior is not correct

### Expected behavior

You are able to drag over a set of rows and make a drag selection successfully. 

For what it's worth, one working version I could find is in `10.6.6` though I haven't checked beyond that,

**Comment by gcko** [comment]
I'll check it out

**Comment by schotime** [comment]
I'm still seeing https://github.com/primefaces/primereact/issues/7221 and have confirmed an issue in 10.8.4.

**Comment by feg3jvl** [comment]
For those who don't need to use onRowMouseDown in the DataTable implementation, the workaround I used to remove the draggable from the tables is this:
```
<PrimeDataTable onMouseDownCapture={(e) => { e.stopPropagation(); }}>
...
```

**Comment by gcko** [comment]
@SolidAnonDev your stackblitz has an issue with the function used for `onSelectionChange` -> changing `e.data` to `e.value` properly enables selection: https://stackblitz.com/edit/vitejs-vite-fqt3cm?file=src%2FApp.tsx

**Comment by SolidAnonDev** [comment]
@gcko I have updated the stackblitz, but am still seeing the same behavior both in the stackblitz and on the official site here:

https://primereact.org/datatable/#multiple_rows_selection

**Comment by SolidAnonDev** [comment]
@gcko I have updated the stackblitz, but am still seeing the same behavior both in the stackblitz and on the official site here:

https://primereact.org/datatable/#multiple_rows_selection

**Comment by SolidAnonDev** [comment]
Sorry, didn't mean to close/reopen.

**Comment by LeandroMarcondes** [comment]
I believe the issue is in the file /datatable/TableBody.js at line 600

The Method **_const onRowMouseDown = (e) => {_**

` event.target.draggable = !isDraggableHandle; // maybe that negation needs to be removed`

I don't understand why event.target.draggable is set to !isDraggableHandle, However, when I remove the negation (!), the selection functionality works again. Additionally, I tested other functions of the datatable, and everything is working correctly.

The last change to this file was in Fix [#6998](https://github.com/primefaces/primereact/pull/7051)

@KumJungMin can you please let me know if this makes sense, and I can proceed with a pull request.

**Comment by KumJungMin** [comment]
> 문제는 /datatable/TableBody.js 파일의 600번째 줄에 있는 것으로 생각됩니다.
> 
> 메서드 **_const onRowMouseDown = (e) => {_**
> 
> ` event.target.draggable = !isDraggableHandle; // maybe that negation needs to be removed`
> 
> event.target.draggable이 !isDraggableHandle로 설정된 이유를 이해할 수 없습니다. 그러나 부정(!)을 제거하면 선택 기능이 다시 작동합니다. 또한 데이터 테이블의 다른 기능을 테스트했고 모든 것이 제대로 작동합니다.
> 
> 이 파일에 대한 마지막 변경 사항은 Fix [#6998 에 있습니다.](https://github.com/primefaces/primereact/pull/7051)
> 
> @KumJungMin이것이 의미가 있는지 알려주시면 풀 리퀘스트를 진행하겠습니다.

---

@LeandroMarcondes 


I refactored the existing code into a concise single line. 
If removing the ! operator works smoothly, you might consider trying that modification. :)
